src/helpers/commands.rs

```diff
@@ -0,0 +1,70 @@
+use crate::ff::FieldType;
+use crate::helpers::{HelperIdentity, Role};
+use crate::protocol::QueryId;
+use std::collections::HashMap;
+
+pub struct Command<'a> {
+    pub dest: &'a HelperIdentity,
+    pub command_type: CommandType<'a>,
+}
+
+impl<'a> Command<'a> {
+    #[must_use]
+    pub fn prepare(dest: &'a HelperIdentity, qc: &'a QueryConfiguration) -> Self {
+        Self {
+            dest,
+            command_type: CommandType::Prepare(qc),
+        }
+    }
+}
+
+pub enum CommandType<'a> {
+    /// Prepare this helper to start processing a new query
+    Prepare(&'a QueryConfiguration<'a>),
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum QueryType {
+    #[cfg(any(test, feature = "test-fixture"))]
+    TestMultiply,
+    IPA,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct RingConfiguration<'a> {
```

Comment:
so the main reason this exists is to enforce that the configuration has 3 entries?

would you just try to use the underlying `assignment`, or do you want to define a `get` fn for `RingConfiguration`?

Reply:
not really, just to be able to extend it later if it needs to carry more information. Can remove if you feel strongly about it

Reply:
actually, i like that it _does_ enforce exactly 3 entries. in fact, maybe it'd be nice to just use position of args to init. something like
```rust
pub fn new(h1_helper_id: HelperIdentity, h2_helper_id: HelperIdentity, h3_helper_id: HelperIdentity) -> Self
```
or something like that